state/execution.go

```diff
@@ -112,14 +112,40 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 	// Tx -> Txs, Message
 	// https://github.com/lazyledger/lazyledger-core/issues/77
 	txs := blockExec.mempool.ReapMaxBytesMaxGas(maxDataBytes, maxGas)
+	l := len(txs)
+	bzs := make([][]byte, l)
+	for i := 0; i < l; i++ {
+		bzs[i] = txs[i]
+	}
+
 	// TODO(ismail):
 	//  1. get those intermediate state roots & messages either from the
 	//     mempool or from the abci-app
 	//  1.1 at this point we should now the square / block size:
 	//      https://github.com/lazyledger/lazyledger-specs/blob/53e5f350838f1e0785ad670704bf91dac2f4f5a3/specs/block_proposer.md#deciding-on-a-block-size
 	//      Here, we instead assume a fixed (max) square size instead.
 	//  2. feed them into MakeBlock below:
-	return state.MakeBlock(height, txs, evidence, nil, nil, commit, proposerAddr)
+
+	processedBlockTxs, err := blockExec.proxyApp.PreprocessTxsSync(abci.RequestPreprocessTxs{Txs: bzs})
+	if err != nil {
+		panic(err) // TODO: what to do??
```

Comment:
What are the error cases that can occur? I think the proposer can not recover from Txs that are in the mem-pool and can not be processed (e.g. due to an encoding problem). In this case, it would be actually OK to panic.